### PR TITLE
docs(community): write a guide for packaging Falcon

### LINKED ISSUE #2081: Write a packager's guide
It seems that people packaging Falcon for various OSes such as Linux distributions (or just hobby projects) have very different ideas how the process should look like, although PEP 517 was supposed to simplify that.

For instance, how do I test the built binary? It seems some packagers opt to invoke `pytest` directly without parameters and expect it to work, disregarding our `tox` environments completely (see: #2080, #2006 etc).

### PR DESCRIPTION
**Pull request by vytas7**
Here's my first stab at creating the packaging guide :sweat_smile:

Closes #2081.

This guide is not aimed at the packaging mechanics of a specific distribution, but rather general guidelines and ideas from our perspective. For reference, we were considering our own Ubuntu/Debian PPA at some point (see #1061 by @jmvrbanac), but it has never materialized, and unfortunately we don't really have the bandwidth for that kind of project.

We do want to make a packager's life easier as much as we can though (not just giving a :middle_finger: pointing to our CI as some other Python project maintainers do that we won't name here _cough_), and I hope 4.0.0 was a step in the right direction.

We sincerely appreciate the work that you put into maintaining Falcon package for various Linux distributions, Conda Forge, etc. :100:

Let me know if you have any feedback; just jot it down here on the PR, or [drop a message on Discussions or Gitter](https://falcon.readthedocs.io/en/stable/community/help.html#chat).

@carlwgeorge @thomasgoirand @stefanor @tacerus @dvzrv @carlodri @synapticarbors @mweinelt

### CODE REVIEW

**Comment by carlwgeorge** [comment]
Wow.  I'm seriously impressed at how thoughtful and thorough this is.  Thank you so much for taking the time to write it!  I see no problem merging it in the current state.

Some optional suggestions, either for the current PR or for later:

It would be nice for this guide to include a link to the changelog so packagers know where to look out for breaking changes.  Most distributions have update policies about what is and what isn't appropriate for their various release channels.  It's not hard to find, but this would be a great place to mention it as well.

On a related note, it helps packagers to know if older major versions (or branches) are still considered maintained, and for how long.  Some projects explicitly only consider the absolute latest version maintained.  Others maintain the current and previous major version.  Unfortunately most projects provide no guidance at all.  This can be a factor when distributions are considering if an update is appropriate in a particular release channel.  For instance, in EPEL 9 (add-on repo for CentOS/RHEL 9), I have version 3.1.3.  If you still consider version 3 maintained to some extent, I feel more comfortable sticking with that version in that channel long term.  If it is intended that there will never be another version 3 release, even for security fixes, then I'll know that I should be prepared to do an [incompatible upgrade](https://docs.fedoraproject.org/en-US/epel/epel-policy-incompatible-upgrades/) if a security issue arises.

**Comment by vytas7** [comment]
Thank you for your kind words and suggestions @carlwgeorge!
I will add a link to the [changelog](https://falcon.readthedocs.io/en/stable/changes/index.html), and assure that we strictly follow SemVer.

Your second point is trickier; I'm afraid we don't have a definite answer. This is very good feedback, and I agree that we ought to have a clear policy on what is considered maintained (both actively, and security updates only), and for how long. We'll discuss this, and try to formulate it. Given our limited resources and (currently) the lack of corporate backing, you might guess which decision this gravitates toward, but we'll see.

**Edit**: I've captured this in a separate issue: #2410.

**Update**: @carlwgeorge  #2410 has now been addressed. We have defined the following policy: [Supported Releases](https://falcon.readthedocs.io/en/latest/community/releases.html#supported-releases).

**Review by CaselIT — COMMENTED** [review]
left some comments. overall it's fine!

File: docs/community/packaging.rst [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+Normally, the recommended way to :ref:`install Falcon <install>` into your
+project is using ``pip`` (or a compatible package/project manager, such as
+``poetry``, ``uv``, and many others) that fetches package archives from
+`PyPI`_.
+
+However, the `PyPI`_-based way of installation is not always applicable or
+optimal. For instance, when the system package manager of a Linux distribution
+is used to install Python-based software, it normally gets the dependencies
+from the same distribution channel, as the specific versions of the packages
+(often not without minor patches) were carefully tested to work well together
```
[the remark this answers is not in the document]
Comment (vytas7) [reply docs/community/packaging.rst:13]: Hm, I hoped that the meaning I tried to convey was clear. But on consideration, that phrase is not really essential, I just removed it completely.

File: docs/community/packaging.rst [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+It is quite uncommon to also include offline documentation (or to provide a
+separate documentation package) as the user can simply browse our documentation
+at `Read the Docs <https://falcon.readthedocs.io/>`__. Even if the package does
+not contain the latest version of Falcon, it is possible to switch to an
+older one using Read the Docs version picker.
+
+If you do decide to ship the offline docs too, you can build it using
+``docs/Makefile`` (you can also invoke ``sphinx-build`` directly):
+
+* To build HTML docs, use ``make html``.
```
[the remark this answers is not in the document]
Comment (vytas7) [reply docs/community/packaging.rst:180]: Aha, nice catch!

**Review by CaselIT — DISMISSED** [review]

**Review by CaselIT — APPROVED** [review]

**Comment by vytas7** [comment]
Merged, the rendered version is now available at https://falcon.readthedocs.io/en/latest/community/packaging.html (the unstable, "latest" version of our docs).

**Pull request merged**